### PR TITLE
Modify S5547(PL/SQL): Migrate to LayC - code re-creation

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Comment by gaetan-ferry-sonarsource** [comment]
beware, asciidoc validation is failing.

File: rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+== How to fix it in  ``++DBMS_CRYPTO++``
```
Comment (loris-s-sonarsource) [inline rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc:1]:
```suggestion
== How to fix it in ``++DBMS_CRYPTO++``
```

File: rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                           + DBMS_CRYPTO.CHAIN_CBC
+                           + DBMS_CRYPTO.PAD_PKCS5;
+----
+
+==== Compliant solution
+
+[source,sql,diff-id=1,diff-type=compliant]
+----
+PLS_INTEGER := DBMS_CRYPTO.ENCRYPT_AES256
+                           + DBMS_CRYPTO.CHAIN_CBC
```
Comment (gaetan-ferry-sonarsource) [inline rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc:21]: I think it is not very good to advise CBC here considering we discourage using it in another rule. But apparently DBMS crypto only supports old school block cipher mode of operation.

File: rules/S5547/plsql/rule.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+include::../impact.adoc[]
+
+// How to fix it section
+
+include::how-to-fix-it/dbms-crypto.adoc[]
+
+== Resources
+
+include::../common/resources/standards.adoc[]
+
```
Comment (gaetan-ferry-sonarsource) [inline rules/S5547/plsql/rule.adoc:16]: I think it would make sense to add a Documentation entry to the DBMS Crypto documentation.

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM. I suggest adding a documentation entry for the DBMS crypto documentation.

File: rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc (hunk 2 above)
Comment (loris-s-sonarsource) [reply rules/S5547/plsql/how-to-fix-it/dbms-crypto.adoc:21]: Yeah, in general we try to mention GCM in these cases. Do you think I should actually remove the CHAIN_* and PAD_* on both examples, or is keeping them better than nothing?

File: rules/S5547/plsql/rule.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+include::../rationale.adoc[]
+
+include::../impact.adoc[]
+
+// How to fix it section
+
+include::how-to-fix-it/dbms-crypto.adoc[]
+
+== Resources
+
```
Comment (loris-s-sonarsource) [inline rules/S5547/plsql/rule.adoc:14]:
```suggestion

=== Documentation

* Oracles docs - https://docs.oracle.com/en/database/oracle/oracle-database/21/arpls/DBMS_CRYPTO.html[DBMS_CRYPTO reference]
```

File: rules/S5547/plsql/rule.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+=== Documentation
+
+* Oracles docs - https://docs.oracle.com/en/database/oracle/oracle-database/21/arpls/DBMS_CRYPTO.html[DBMS_CRYPTO reference]
```
Comment (loris-s-sonarsource) [inline rules/S5547/plsql/rule.adoc:17]:
```suggestion
* Oracles docs - https://docs.oracle.com/en/database/oracle/oracle-database/21/arpls/DBMS_CRYPTO.html[DBMS_CRYPTO reference]

```

**Pull request merged**